### DISCUSSION TITLE
How can i record velocity message?

### DISCUSSION THREAD

**Discussion by lwisle** [USC-ACTLab/crazyswarm#612] [Q&A] [answered]
I found if I record message by rosbag, I can only get position message. If I want to get more details about crazyflie such as velocity, what should i do? Which file should i modify to set a topic to subscribe velocity message? Thank you !
![2022-03-12 15-13-46屏幕截图](https://user-images.githubusercontent.com/82260604/158008198-31d2bff8-a2e4-4aa1-b479-a6bc8d67d70d.png)

**Comment by knmcguire** [comment, marked as answer]
Hi! Have you looked at the logging functionality[ here in the launch file](https://github.com/USC-ACTLab/crazyswarm/blob/60a34390d17fc327a14989fe5459dcd96a53ebe6/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L10-L13)? 

You can select which generic logging variables you want to log from the crazyflie firmware, that are described here: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimate

don't forget to put  enable_logging parameter on true in the launch file.

**Reply by lwisle** [reply]
Thanks for your help！I followed your guidance and recorded many messages,but some problems occured when i logged stateEstimate.vz.
How can I get this message? Thanks.
![3d18331d883dcbcb0856c172c3e9c87](https://user-images.githubusercontent.com/82260604/160222079-3aa85a2d-aa06-490a-8cdb-4dfb87308596.jpg)

**Reply by knmcguire** [reply]
That is quite strange... stateEstimate.vz is a core logging variable so it should always be in toc.

Could you double check the version of your crazyflie firmware? Have you updated the crazyradio firmware? You can also check if you can log that variable with the cflib example for basic logging: https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/logging/basiclog.py

**Reply by lwisle** [reply]
when I use cflib with cfclient, I can log stateEstimate.vz. And the version of firmware is the latest. 
What's more, I can log stateEstimate.x,stateEstimate.vx,stateEstimate.y,stateEstimate.vy and stateEstimate.z with crazyswarm.
![图片](https://user-images.githubusercontent.com/82260604/160744917-812e8a9c-3f22-4e00-b8ab-18c0075320f5.png)
Here is the list when I use `rosrun crazyflie_tools listLogVariables`
![图片](https://user-images.githubusercontent.com/82260604/160746835-1c993790-ada0-45cd-acdc-c17e6e623096.png)

**Reply by knmcguire** [reply]
And if you remove all the other variables for the exception of that vz variables in the launch file? You might be logging too much at the same time as these log blocks do have a size limit.

**Reply by lwisle** [reply]
Actually I failed even if I only logged stateEstimate.vz. But now I use stateEstimateZ.vz stead. Thanks for your help

**Reply by knmcguire** [reply]
ah oke, yeah that will work though. It's still an issue but at least you are not blocked for your work anymore
